quickwit/quickwit-ingest/src/ingest_v2/mrecord.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use bytes::{Buf, Bytes};
+use prost::Message;
+use quickwit_proto::ingest::{m_record_header, MRecordHeader};
+
+pub(super) enum MRecordPayload {
+    Doc(Bytes),
+    Commit,
+}
+
+pub(super) struct MRecord {
```

Comment:
I am not fond of the implicitly entangle double `enums` (MRecordPayload and MRecordHeader)

I would suggest
```
enum MRecord {
      Doc(DocRecord),
      Commit(CommitHeader),
}

struct DocRecord {
    header: DocHeader,
    payload: Bytes, 
}
```

or if the intermediary struct does not help (as in, is not useful in other methods prototypes)

```
enum MRecord {
      Doc {
           header: DocHeader,
           payload: Bytes,
      },
      Commit(CommitHeader),
}
```